test/webrtc_test.rb

```diff
@@ -0,0 +1,189 @@
+            emit task2.start_event
+        end
+    end
+
+    it "emits start only after the data channel is fully established" do
+        deploy("task_a", "task_b", Time.at(2))
+        configure_and_start
+    end
+
+    it "transmits data from the active to the passive task once started" do
+        deploy("task_a", "task_b", Time.at(5))
+        configure_and_start
+
+        data_in = Types.iodrivers_base.RawPacket.new
+        data_in.time = Time.now
+        data_in.data = [1, 0, 1, 0, 1, 1, 1, 0]
+
+        data = expect_execution do
+            syskit_write task1.data_in_port, data_in
+        end.to { have_one_new_sample task2.data_out_port }
+
+        assert_equal data_in.data, data.data
+    end
+
+    it "transmits data from the passive to the active task once started" do
+        deploy("task_a", "task_b", Time.at(5))
+        configure_and_start
+
+        data_in_task1 = Types.iodrivers_base.RawPacket.new
+        data_in_task1.time = Time.now
+        data_in_task1.data = [1, 0, 1, 0, 1, 1, 1, 0]
+
+        expect_execution do
+            syskit_write task1.data_in_port, data_in_task1
+        end.to { have_one_new_sample task2.data_out_port }
+
+        data_in_task2 = Types.iodrivers_base.RawPacket.new
+        data_in_task2.time = Time.now
+        data_in_task2.data = [0, 0, 0, 0, 1, 0, 1, 0]
+
+        output = expect_execution do
+            syskit_write task2.data_in_port, data_in_task2
+        end.to { have_one_new_sample task1.data_out_port }
+
+        assert_equal data_in_task2.data, output.data
+    end
+
+    it "successfully reestablishes connection after a stop/start cycle" do
+        deploy("task_a", "task_b", Time.at(5))
+        configure_and_start
+
+        data_in = Types.iodrivers_base.RawPacket.new
+        data_in.time = Time.now
+        data_in.data = [1, 0, 1, 0, 1, 1, 1, 0]
+
+        data = expect_execution do
+            syskit_write task1.data_in_port, data_in
+        end.to { have_one_new_sample task2.data_out_port }
+
+        assert_equal data_in.data, data.data
+
+        state_task = Types.comms_webrtc.WebRTCState.new
+        state_task.peer_connection.state = :NoConnection
+        state_task.peer_connection.gathering_state = :NoGathering
+        state_task.peer_connection.signaling_state = :NoSignaling
+        state_task.data_channel = :NoDataChannel
+        state_task.web_socket = :NoWebSocket
+
+        state = expect_execution do
+            task1.stop!
+            task2.stop!
+        end.to do
+            emit task1.stop_event
+            emit task2.stop_event
+            [have_one_new_sample(task1.status_port),
+             have_one_new_sample(task2.status_port)]
+        end
+
+        assert_equal state_task, state[0]
+        assert_equal state_task, state[1]
+
+        deploy("task_c", "task_d", Time.at(5))
+        configure_and_start
+    end
+
+    it "returns exception when the wait to find the remote peer id exceeds the timeout" do
```

Comment:
You can also add a test for the websocket connection timeout by passing 240.24.15.234 as the signalling server IP, and trying to configure it (see https://www.rock-robotics.org/rock-and-syskit/components/testing.html#testing-configuration-or-start-failures for that last bit)